stock_picking_group_by_partner_by_carrier_by_date/models/stock_move.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Camptocamp (https://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _domain_search_picking_for_assignation(self):
+        domain = super()._domain_search_picking_for_assignation()
+        by_date_domain = self._domain_search_picking_for_assignation_by_date()
+        domain.extend(by_date_domain)
+        return domain
+
+    def _skip_search_picking_for_assignation_by_date(self):
+        return (
+            not self.picking_type_id.group_pickings_by_date
+            or not self.picking_type_id.group_pickings
+            or self.group_id.sale_id.picking_policy == "one"
+        )
+
+    def _domain_search_picking_for_assignation_by_date(self):
+        domain = []
+        if self._skip_search_picking_for_assignation_by_date():
+            return domain
+        date_planned = self.date_expected
+        date_planned_end = date_planned.replace(hour=23, minute=59, second=59)
```

Comment:
This is not timezone safe. date_planned is in UTC. It should be end of day in the timezone of the warehouse.
Something like:
tz_date_planned = date_planned.astimezone(tz_wh)
tz_date_planned_end = tz_date_planned.replace(hour=23, minute=59, second=59)
date_planned_end = tz_utils.tz_to_utc_naive_datetime(tz_wh, tz_date_planned_end)

Reply:
@jbaudoux when the cutoff module is installed all expected dates will be set considering the wh tz already.
Here I'm using a date that is already matching its own WH TZ. Am I wrong?

Reply:
As discussed, I'll approve this PR as you prepare the fix in another coming PR